utils/oval/oval.go

```diff
@@ -0,0 +1,426 @@
+	//Namespace stores the namespace that will be used in clair to store the vulnerabilities
+	Namespace() string
+}
+
+var (
+	log = capnslog.NewPackageLogger("github.com/coreos/clair", "utils/oval")
+)
+
+// FetchUpdate gets vulnerability updates from the OVAL definitions.
+func (f *OvalFetcher) FetchUpdate(datastore database.Datastore) (resp updater.FetcherResponse, err error) {
+	log.Info("fetching %s vulnerabilities", f.OsInfo.DistName())
+
+	if err != nil {
+		return resp, err
+	}
+
+	r, err := http.Get(f.OsInfo.OvalURI())
+
+	if err != nil {
+		log.Errorf("could not download %s's update list: %s", f.OsInfo.DistName(), err)
+		return resp, cerrors.ErrCouldNotDownload
+	}
+
+	var distList []string
+	scanner := bufio.NewScanner(r.Body)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		filename := f.OsInfo.ParseFilenameDist(line)
+		if filename != "" {
+			distList = append(distList, filename)
+		}
+	}
+
+	for _, filename := range distList {
+		r, err := http.Get(filename)
+
+		if err != nil {
+			log.Errorf("could not download %s's update file: %s", f.OsInfo.DistName(), err)
+			return resp, cerrors.ErrCouldNotDownload
+		}
+
+		vs, err := f.ParseOval(r.Body)
+
+		if err != nil {
+			return resp, err
+		}
+
+		for _, v := range vs {
+			resp.Vulnerabilities = append(resp.Vulnerabilities, v)
+		}
+	}
+
+	// Set the flag if we found anything.
+	if len(distList) > 0 {
+		resp.FlagName = f.OsInfo.DistName() + "updater"
+		resp.FlagValue = distList[len(distList)-1]
+	} else {
+		log.Debug("no #{f.OsInfo.DistroName()}")
+	}
+
+	return resp, nil
+}
+
+// Clean deletes any allocated resources.
+func (f *OvalFetcher) Clean() {}
+
+// Parse criterions into an array of FeatureVersion for storing into the database
+func (f *OvalFetcher) ToFeatureVersions(possibilities [][]criterion) []database.FeatureVersion {
+	featureVersionParameters := make(map[string]database.FeatureVersion)
+
+	for _, criterions := range possibilities {
+		var (
+			featureVersion database.FeatureVersion
+			osVersion      string
+		)
+
+		for _, c := range criterions {
+			tmp_v := f.OsInfo.ParseOsVersion(c.Comment)
+			if tmp_v != "" {
+				osVersion = tmp_v
+				continue
+			}
+
+			tmp_p_name, tmp_p_version := f.OsInfo.ParsePackageNameVersion(c.Comment)
+			if tmp_p_version != "" && tmp_p_name != "" {
+				featureVersion.Feature.Name = tmp_p_name
+				featureVersion.Version, _ = types.NewVersion(tmp_p_version)
+				continue
+			}
+
+			log.Warningf("could not parse criteria: '%s'.", c.Comment)
+		}
+
+		if osVersion == "" {
+			log.Warning(criterions)
+			continue
+		}
+
+		featureVersion.Feature.Namespace.Name = fmt.Sprintf("%s:%s", f.OsInfo.Namespace(), osVersion)
+
+		if featureVersion.Feature.Name != "" && featureVersion.Version.String() != "" {
+			featureVersionParameters[featureVersion.Feature.Namespace.Name+":"+featureVersion.Feature.Name] = featureVersion
+		} else {
+			log.Warningf("could not determine a valid package from criterions: %v", criterions)
```

Comment:
Is the value of %v for criterions useful? Is there a useful identifier for each section of criterions so that someone could refer to the file and know where the failure was?